File: sdk/src/main/java/com/meniga/sdk/webservices/requests/UpdateBudget.java
package com.meniga.sdk.webservices.requests;

import java.util.List;

public class UpdateBudget extends QueryRequestObject {

    public String name;
    public String description;
    public List<Long> accountIds;

    @Override
    public long getValueHash() {
        int result = name != null ? name.hashCode() : 0;
        result = 31 * result + (description != null ? description.hashCode() : 0);
        result = 31 * result + (accountIds != null ? accountIds.hashCode() : 0);
        return result;
    }
}

File: sdk/src/main/java/com/meniga/sdk/webservices/budget/UpdateBudget.kt
/*
 * Copyright 2017 Meniga Iceland Inc.
 */
package com.meniga.sdk.webservices.budget

import com.meniga.sdk.webservices.requests.QueryRequestObject

data class UpdateBudget(
        var name: String? = null,
        var description: String? = null,
        var accountIds: List<Long>? = null
) : QueryRequestObject() {

    override fun getValueHash(): Long = hashCode().toLong()
}
